Userland/Games/ColorLines/ColorLines.cpp

```diff
@@ -0,0 +1,402 @@
+    result.append(MUST(result.last()->rotated(Gfx::RotationDirection::Clockwise)));
+    return result;
+}
+
+ColorLines::ColorLines(StringView app_name)
+    : m_app_name { app_name }
+    , m_game_state { GameState::Idle }
+    , m_board { make<MarbleBoard>() }
+    , m_marble_bitmaps { build_marble_color_bitmaps() }
+    , m_trace_bitmaps { build_marble_trace_bitmaps() }
+    , m_score_font { Gfx::BitmapFont::load_from_file("/res/fonts/MarietaBold24.font") }
+{
+    VERIFY(m_marble_bitmaps.size() == Marble::number_of_colors);
+    set_font(Gfx::FontDatabase::default_fixed_width_font().bold_variant());
+    m_high_score = Config::read_i32(m_app_name, m_app_name, "HighScore"sv, 0);
+    reset();
+}
+
+void ColorLines::reset()
+{
+    set_game_state(GameState::StartingGame);
+}
+
+void ColorLines::mousedown_event(GUI::MouseEvent& event)
+{
+    if (m_game_state != GameState::Idle && m_game_state != GameState::MarbleSelected)
+        return;
+    auto const event_position = event.position().translated(
+        -frame_inner_rect().x(),
+        -frame_inner_rect().y() - board_vertical_margin);
+    if (event_position.x() < 0 || event_position.y() < 0)
+        return;
+    auto const clicked_cell = Point { event_position.x() / board_cell_dimension.width(),
+        event_position.y() / board_cell_dimension.height() };
+    if (!MarbleBoard::in_bounds(clicked_cell))
+        return;
+    if (m_board->has_selected_marble()) {
+        auto const selected_cell = m_board->selected_marble().position();
+        if (selected_cell == clicked_cell) {
+            m_board->reset_selection();
+            set_game_state(GameState::Idle);
+            return;
+        }
+        if (m_board->is_empty_cell_at(clicked_cell)) {
+            if (m_board->build_marble_path(selected_cell, clicked_cell, m_marble_path))
+                set_game_state(GameState::MarbleMoving);
+            return;
+        }
+        if (m_board->select_marble(clicked_cell))
+            set_game_state(GameState::MarbleSelected);
+        return;
+    }
+    if (m_board->select_marble(clicked_cell))
+        set_game_state(GameState::MarbleSelected);
+}
+
+void ColorLines::timer_event(Core::TimerEvent&)
+{
+    switch (m_game_state) {
+    case GameState::GeneratingMarbles:
+        update();
+        if (--m_marble_animation_frame < animation_frame_marble_generating_end) {
+            m_marble_animation_frame = default_marble_animation_frame;
+            set_game_state(GameState::CheckingMarbles);
+        }
+        break;
+
+    case GameState::MarbleSelected:
+        m_marble_animation_frame = (m_marble_animation_frame + 1) % number_of_marble_bounce_frames;
+        update();
+        break;
+
+    case GameState::MarbleMoving:
+        m_marble_animation_frame = (m_marble_animation_frame + 1) % number_of_marble_bounce_frames;
+        update();
+        if (m_marble_path.remaining_steps() != 1 && m_marble_animation_frame != 2)
```

Comment:
Please give this magic number `2` a name, like you have with the other important animation frames.